npm installation fails

npm installation fails below versions.

Platform : Windows 10
Node version : 8.4.0
Npm version : 5.3.0
Python version : 2.7.13

```
C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter>npm install

> node-zopfli@1.4.0 install C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download(403): https://node-zopfli.s3.amazonaws.com/Release/zopfli-v1.4.0-node-v57-win32-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for node-zopfli@1.4.0 and node@8.4.0 (node-v57 ABI) (falling back to source compile with node-gyp)
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Platforms\x64\PlatformToolsets\v140\Toolset.targets(36,5): error MSB8036: The Windows SDK version 8.1 was not found. Install
 the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\Development\W
orkspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli\build\zopfli.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-gyp\lib\build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Development\\Workspaces\\TESTS\\Universal\\fusebox-angular-universal-starter\\node_modules\\node-gyp\\bin\\node-gyp.js" "build" "--fallback-to-build" "--module=C:\\Development\\Workspaces\\TESTS\\Universal\\fusebox-angular-universal-starter\\node_modules\\node-zopfli\\lib\\binding\\node-v57-win32-x64\\zopfli.node" "--module_name=zopfli" "--module_path=C:\\Development\\Workspaces\\TESTS\\Universal\\fusebox-angular-universal-starter\\node_modules\\node-zopfli\\lib\\binding\\node-v57-win32-x64"
gyp ERR! cwd C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli
gyp ERR! node -v v8.4.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --module=C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli\lib\binding\node-v57-win32-x64\zopfli.node --module_name=zopfli --module_path=C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli\lib\binding\node-v57-win32-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:125:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:927:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
node-pre-gyp ERR! System Windows_NT 10.0.14393
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Development\\Workspaces\\TESTS\\Universal\\fusebox-angular-universal-starter\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli
node-pre-gyp ERR! node -v v8.4.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.36
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files\nodejs\node.exe C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --module=C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli\lib\binding\node-v57-win32-x64\zopfli.node --module_name=zopfli --module_path=C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-zopfli\lib\binding\node-v57-win32-x64' (1)
npm WARN @angular/cdk@2.0.0-beta.10 requires a peer of @angular/core@^4.3.0 but none was installed.
npm WARN @angular/cdk@2.0.0-beta.10 requires a peer of @angular/common@^4.3.0 but none was installed.
npm WARN @angular/material@2.0.0-beta.10 requires a peer of @angular/core@^4.3.0 but none was installed.
npm WARN @angular/material@2.0.0-beta.10 requires a peer of @angular/common@^4.3.0 but none was installed.
npm WARN @nguniversal/express-engine@1.0.0-beta.3 requires a peer of @angular/core@^4.0.0 but none was installed.
npm WARN @nguniversal/express-engine@1.0.0-beta.3 requires a peer of @angular/platform-server@^4.0.0 but none was installed.
npm WARN @ngx-meta/core@0.4.0-rc.2 requires a peer of @angular/core@^4.0.0 but none was installed.
npm WARN @ngx-meta/core@0.4.0-rc.2 requires a peer of @angular/platform-browser@^4.0.0 but none was installed.
npm WARN @ngx-meta/core@0.4.0-rc.2 requires a peer of @angular/router@^4.0.0 but none was installed.
npm WARN angular2-jwt@0.2.3 requires a peer of @angular/core@^2.0.0||^4.0.0 but none was installed.
npm WARN angular2-jwt@0.2.3 requires a peer of @angular/http@^2.0.0||^4.0.0 but none was installed.
npm WARN codelyzer@3.1.2 requires a peer of @angular/compiler@^2.3.1 || >=4.0.0-beta <5.0.0 but none was installed.
npm WARN codelyzer@3.1.2 requires a peer of @angular/core@^2.3.1 || >=4.0.0-beta <5.0.0 but none was installed.
npm WARN tsickle@0.23.6 requires a peer of typescript@2.3.4 but none was installed.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-zopfli@1.4.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-zopfli@1.4.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\merih\AppData\Roaming\npm-cache\_logs\2017-09-01T14_59_37_299Z-debug.log

C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter>
```

I confirm that npm installation successfully done using @bluelightning2k solution but I didn't make it run using "npm start" command after installation.

```
C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter>npm start

> fusebox-angular-universal-starter@0.0.0-development start C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter
> ts-node fuse.ts serve --build-type dev --env-config dev

Error: ENOENT: no such file or directory, scandir 'C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-sass\vendor'
    at Object.fs.readdirSync (fs.js:910:18)
    at Object.getInstalledBinaries (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-sass\lib\extensions.js:124:13)
    at foundBinariesList (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-sass\lib\errors.js:20:15)
    at foundBinaries (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-sass\lib\errors.js:15:5)
    at Object.module.exports.missingBinary (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-sass\lib\errors.js:45:5)                           at module.exports (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-sass\lib\binding.js:15:30)
    at Object.<anonymous> (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\node-sass\lib\index.js:14:35)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\coffee-script\lib\coffee-script\register.js:45:36)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
    at Module.require (module.js:517:17)                                                                                                                                                     at require (internal/module.js:11:18)                                                                                                                                                    at Object.<anonymous> (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\tools\tasks\seed\sass.ts:5:19)                                                        at Module._compile (module.js:573:30)
    at Module.m._compile (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\ts-node\dist\index.js:219:29)
    at Module._extensions..js (module.js:584:10)
    at Object.require.extensions.(anonymous function) [as .ts] (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\ts-node\dist\index.js:221:16)       at Module.load (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\coffee-script\lib\coffee-script\register.js:45:36)                              at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)                                                                                                                                               at Module.require (module.js:517:17)                                                                                                                                                     at require (internal/module.js:11:18)
    at C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\tools\tasks\index.ts:5:83                                                                                 at Array.forEach (<anonymous>:null:null)
    at Object.<anonymous> (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\tools\tasks\index.ts:5:50)
    at Module._compile (module.js:573:30)
    at Module.m._compile (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\ts-node\dist\index.js:219:29)
    at Module._extensions..js (module.js:584:10)
    at Object.require.extensions.(anonymous function) [as .ts] (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\ts-node\dist\index.js:221:16)       at Module.load (module.js:507:32)                                                                                                                                                        at tryModuleLoad (module.js:470:12)                                                                                                                                                      at Function.Module._load (module.js:462:3)
    at Module.require (module.js:517:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\fuse.ts:19:1)                                                                         at Module._compile (module.js:573:30)                                                                                                                                                    at Module.m._compile (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\ts-node\dist\index.js:219:29)
    at Module._extensions..js (module.js:584:10)
    at Object.require.extensions.(anonymous function) [as .ts] (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\ts-node\dist\index.js:221:16)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)                                                                                                                                                      at Function.Module._load (module.js:462:3)                                                                                                                                               at Function.Module.runMain (module.js:609:10)
    at Object.<anonymous> (C:\Development\Workspaces\TESTS\Universal\fusebox-angular-universal-starter\node_modules\ts-node\dist\_bin.js:119:16)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
    at Function.Module.runMain (module.js:609:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:598:3

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fusebox-angular-universal-starter@0.0.0-development start: `ts-node fuse.ts serve --build-type dev --env-config dev`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fusebox-angular-universal-starter@0.0.0-development start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\merih\AppData\Roaming\npm-cache\_logs\2017-09-02T23_53_38_860Z-debug.log
```

--- a/package.json
+++ b/package.json
@@ -47,8 +47,10 @@
     "@compodoc/compodoc": "^1.0.0-beta.14",
     "@types/browser-sync": "^0.0.36",
     "@types/bunyan": "^1.8.2",
+    "@types/compression": "0.0.34",
     "@types/cookie-parser": "^1.4.1",
     "@types/express": "^4.0.37",
+    "@types/getos": "^3.0.0",
     "@types/glob": "^5.0.32",
     "@types/html-minifier": "^1.1.30",
     "@types/jest": "^20.0.8",
@@ -124,17 +126,21 @@
     "cookie-parser": "^1.4.3",
     "core-js": "2.5.1",
     "express": "^4.15.4",
+    "getos": "^3.0.1",
     "hammerjs": "^2.0.8",
     "js-cookie": "^2.1.4",
     "morgan": "^1.8.2",
     "ms": "^2.0.0",
     "rxjs": "^5.4.3",
     "serve-favicon": "^2.4.3",
-    "shrink-ray": "^0.1.3",
     "stacktrace-js": "^2.0.0",
     "systemjs": "^0.20.18",
     "tsickle": "^0.23.6",
     "zone.js": "^0.8.17"
+  },
+  "optionalDependencies": {
+    "compression": "^1.7.0",
+    "shrink-ray": "^0.1.3"
   },
   "engines": {
     "node": ">= 8.4.0",

--- a/src/server/server.ts
+++ b/src/server/server.ts
@@ -7,6 +7,8 @@ import * as express from 'express'
 import * as morgan from 'morgan'
 import * as favicon from 'serve-favicon'
 import * as cookieParser from 'cookie-parser'
+import * as getOs from 'getos'
+import * as compression from 'compression'
 import ms = require('ms')
 import { createLogger } from '@expo/bunyan'
 import { join, resolve } from 'path'
@@ -51,7 +53,16 @@ app.engine('html', ngExpressEngine({ bootstrap: AppServerModule }))
 app.set('view engine', 'html')
 app.set('views', root)
 app.use(cookieParser())
-app.use(shrinkRay())
+
+getOs((err, os) => {
+  if (err) throw err
+  if (os.os === 'win32') {
+    app.use(compression())
+  } else {
+    app.use(shrinkRay())
+  }
+})
+
 if (__process_env__.server.minifyIndex) {
   app.use(minifyHTML({
     override: true,

